compiler/rustc_mir_transform/src/elaborate_drops.rs

```diff
@@ -121,13 +120,27 @@ fn find_dead_unwinds<'tcx>(
             maybe_live |= flow_inits.contains(child);
         });
 
-        debug!("find_dead_unwinds @ {:?}: maybe_live={}", bb, maybe_live);
+        debug!("remove_dead_unwinds @ {:?}: maybe_live={}", bb, maybe_live);
         if !maybe_live {
             dead_unwinds.insert(bb);
         }
     }
 
-    dead_unwinds
+    if dead_unwinds.is_empty() {
+        return;
+    }
+
+    for (bb, bb_data) in body.basic_blocks_mut().iter_enumerated_mut() {
+        match bb_data.terminator_mut().kind {
+            TerminatorKind::Drop { ref mut unwind, .. }
+            | TerminatorKind::DropAndReplace { ref mut unwind, .. }
+                if unwind.is_some() && dead_unwinds.contains(bb) =>
+            {
+                *unwind = None;
+            }
+            _ => continue,
+        };
+    }
```

Comment:
can't we patch the unwind target inside the loop directly?

Reply:
Data flow results cursor borrows the MIR body, so modification needs to be a separate step.

Reply:
Ah yes, thanks.

nit: not sure how much this would impact perfs, but wouldn't iterating over `dead_unwinds` consider only the interesting parts? If we have few of those blocks we might save passing over the whole body.

Something like
```
for bb in dead_unwinds.iter() {
    match body.basic_blocks_mut()[bb].terminator_mut().kind {
      ...
   }
}
```